tests/CMakeLists.txt

```diff
@@ -216,3 +216,16 @@ add_test(NAME connection-error-test
          COMMAND "${CMAKE_SOURCE_DIR}/tests/scripts/connection-error-test.sh"
                  "$<TARGET_FILE:clusterclient_async_sequence>"
          WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/tests/scripts/")
+add_test(NAME redirect-with-hostname-test
+         COMMAND "${CMAKE_SOURCE_DIR}/tests/scripts/redirect-with-hostname-test.sh"
+                 "$<TARGET_FILE:clusterclient>"
+         WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/tests/scripts/")
+# This test can't be run on hiredis v1.1.0 due to hiredis issue #1171.
+add_test(NAME redirect-with-hostname-test-async
+         COMMAND "${CMAKE_SOURCE_DIR}/tests/scripts/redirect-with-hostname-test.sh"
+                 "$<TARGET_FILE:clusterclient_async_sequence>"
+         WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/tests/scripts/")
+# Disable testcase if hiredis contains the issue, or if the version is unknown.
+if(hiredis_VERSION VERSION_EQUAL "1.1.0" OR hiredis_VERSION VERSION_EQUAL "0")
```

Comment:
Actually there is [VERSION_LESS](https://cmake.org/cmake/help/latest/command/if.html#version-less)

Reply:
For `1.0.X` the testcase works and will be run with this check, just like `1.1.1`.

If CMake cant figure out which hiredis version that is installed then the version is interpreted as "0".
This is the case when CMake is downloading hiredis itself funny enough..some problem with its command FetchContent.

There seems to be problems building hiredis-cluster with hiredis 0.x.y when testing,
`hi_free` is not defined by hiredis since it looks to be introduced in 1.0.

Reply:
Right, the test case is not using the PREFER_IPV4,IPV6 options, the ones that introduced the bug, so versions without these are ok.

For the 0.x.y problems, I guess we require 1.0 for hiredis-cluster. We should just document it, if it's not already done.